### ISSUE TITLE
Skin Lesion Segmentation

### ISSUE THREAD

**Issue by jcwang123** [HiLab-git/CA-Net#3]
In the paper, it's written that the images are randomly divided into three subsets for training, validating and testing, respectively.
While in this repo, authors perform 5-folds cross-validation.
So, which one is the true setting presented in the paper?

**Comment by ElegantLin** [comment]
Hi, @jcwang123  I am sorry to turn to you in this repo. I met some problems when I am running https://github.com/krishnabits001/domain_specific_cl. I cannot reproduce the results in the paper. Could I ask you some details about this?

Thanks a lot!

**Comment by cjtcn** [comment]
I also want to know how to do  5-folds cross-validation, do you understand it now?

**Comment by jcwang123** [comment]
It seems that there is an additional validation set when performing 5-folds cross-validation.

**Comment by cjtcn** [comment]
> It seems that there is an additional validation set when performing 5-folds cross-validation.

I don't understand,  all images are randomly divided into three sets for training, validating and testing, what is the additional validation set?

**Comment by jcwang123** [comment]
Firstly, all images are randomly divided into 5 folds.
Secondly, in each fold, images are randomly divided into training set and validation set.

**Comment by cjtcn** [comment]
> Firstly, all images are randomly divided into 5 folds. Secondly, in each fold, images are randomly divided into training set and val

> Firstly, all images are randomly divided into 5 folds. Secondly, in each fold, images are randomly divided into training set and validation set.

Firstly, each fold have all 2594 images.
Secondly, in each fold, images are randomly divided into train ,validation, test sets. 
Do i understand right?
 can you tell me how to 5-folds cross-validation in detail? Thanks for that.

**Comment by jcwang123** [comment]
For example, let fd1...5 denote the five parts of images.
In each experiment fold, four folds are used to form the training set and the rest one is used to form the testing set.
You can refer \url{https://github.com/jcwang123/BA-Transformer/blob/main/dataset/isbi2018.py} for details. 

here, authors further dive the training set into training and validating set, different of my codes.

**Comment by lixiangqi400** [comment]
I think the author did not use the 5-fold crossover method, but scrambled the data set into five data sets, each data set has all 2594 pictures, trained five times, and then averaged
